modules/public/public.go

```diff
@@ -18,28 +22,36 @@ import (
 
 // Options represents the available options to configure the handler.
 type Options struct {
-	Directory   string
-	Prefix      string
 	CorsHandler func(http.Handler) http.Handler
 }
 
 // AssetsURLPathPrefix is the path prefix for static asset files
 const AssetsURLPathPrefix = "/assets/"
 
+func CustomAssets() *assetfs.Layer {
+	return assetfs.Local("custom", setting.CustomPath, "public")
+}
+
+func StaticAssets() *assetfs.Layer {
+	return assetfs.Local("static", setting.StaticRootPath, "public")
+}
+
 // AssetsHandlerFunc implements the static handler for serving custom or original assets.
 func AssetsHandlerFunc(opts *Options) http.HandlerFunc {
-	custPath := filepath.Join(setting.CustomPath, "public")
-	if !filepath.IsAbs(custPath) {
-		custPath = filepath.Join(setting.AppWorkPath, custPath)
-	}
-	if !filepath.IsAbs(opts.Directory) {
-		opts.Directory = filepath.Join(setting.AppWorkPath, opts.Directory)
-	}
-	if !strings.HasSuffix(opts.Prefix, "/") {
-		opts.Prefix += "/"
+	var assetFS http.FileSystem
+	if setting.HasBuiltinBindata {
+		assetFS = assetfs.Layered(CustomAssets(), StaticAssets(), BuiltinAssets()) // old behavior: always include StaticAssets
```

Comment:
Looks like it's not the logic I know.

Reply:
The old logic is this:

![image](https://user-images.githubusercontent.com/2114189/230776517-7dbbfb29-18d1-4840-aecc-552d08504eb0.png)

![image](https://user-images.githubusercontent.com/2114189/230776539-cc19f0a0-f7c8-4e07-bafe-aeac09d94169.png)